ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -342,6 +349,36 @@ private static void throwOnNonExecutableStatement(final ConfiguredStatement<?> s
     }
   }
 
+  private static void throwOnUnsupportedKeyFormat(final ConfiguredStatement<?> statement) {
+    try {
+      if (statement.getStatement() instanceof CreateSource) {
+        final CreateSource createSource = (CreateSource) statement.getStatement();
+        throwOnUnsupportedKeyFormat(
+            SourcePropertiesUtil.getKeyFormat(createSource.getProperties()));
+      }
+
+      if (statement.getStatement() instanceof CreateAsSelect) {
+        final CreateAsSelect createAsSelect = (CreateAsSelect) statement.getStatement();
+        createAsSelect.getProperties().getKeyFormat()
+            .ifPresent(EngineExecutor::throwOnUnsupportedKeyFormat);
+      }
+    } catch (KsqlStatementException e) {
+      throw e;
+    } catch (KsqlException e) {
+      throw new KsqlStatementException(
+          ErrorMessageUtil.buildErrorMessage(e),
+          statement.getStatementText(),
+          e.getCause());
+    }
```

Comment:
You're calling `throwOnUnsupportedKeyFormat` from within the `plan` method's try/catch block already, so I don't think you need this try/catch, just let the exception bubble up.

Reply:
And for the record... `ErrorMessageUtil`... yuck!!!! I'd avoid using this class. It's one that should be shot in the head at some point.

Reply:
> You're calling throwOnUnsupportedKeyFormat from within the plan method's try/catch block already, so I don't think you need this try/catch, just let the exception bubble up.

Good call. Made the change.

> And for the record... ErrorMessageUtil... yuck!!!! I'd avoid using this class. It's one that should be shot in the head at some point.

My use of ErrorMessageUtil was just copy-pasta from other parts of the code but I'm curious what your gripe with it is. Is it not useful to reveal the chain of causation to the user?